src/database/entities/event.ts

```diff
@@ -19,22 +39,33 @@ import { NotablePersonEventComment } from './comment';
 export class NotablePersonEvent extends BaseEntity {
   @PrimaryGeneratedColumn('uuid') id: string;
 
-  @Trim()
+  @ValidateIf((_, v) => typeof v === 'string')
   @IsNotEmpty()
-  @Column({ type: 'text', nullable: false })
-  quote: string;
+  @Column({ type: 'text', nullable: true })
+  quote: string | null;
 
-  @Column({ nullable: false })
-  isQuoteByNotablePerson: boolean;
+  @Column({ nullable: true, type: 'tinyint' })
+  isQuoteByNotablePerson: boolean | null;
 
-  @IsUrl({
-    require_protocol: true,
-    require_valid_protocol: true,
-    protocols: ['https', 'http'],
+  @Column({
+    nullable: false,
+    type: 'enum',
+    default: null,
+    enum: Object.keys(eventTypes),
   })
+  type: EventType;
+
+  @IsUrl(urlValidationOptions)
   @Trim()
   @Column({ type: 'text', nullable: false })
-  sourceUrl: string;
+  sourceUrl: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  entityUrl: string | null;
```

Comment:
What's the difference between `entityUrl` and `sourceUrl`? Are we ever going to have/accept an event without a `sourceUrl`?

Reply:
I'm following the design here. `entityUrl` is the website of the organization/campaign/event. `sourceUrl` is the website that reported the event.

Reply:
For now `sourceUrl` is required. Do you think we should relax this?

Reply:
I didn't know that we accept `entityUrl`. But if it's the URL for the website of organization/campaign/event, why is it called `entityUrl`? Entity URL sounds like it's a URL to the database object or something?

With regard to `sourceUrl`, I don't think we should accept an event without a source. We'll always need a source. But you made it nullable, so I thought you were making it optional?

Reply:
Right. This is a mistake. In GraphQL schema, it's non-nullable. I'll fix this.

Any suggestion to rename `entityUrl`?

Reply:
One option could be `organizationWebsiteUrl`?